### DISCUSSION TITLE
How to use winget to uninstall a software?

### DISCUSSION THREAD

**Discussion by Foadsf** [microsoft/winget-cli#3131] [Q&A] [answered]
When I run the `winget upgrade` command, it shows a list of upgradeable packages including

> Microsoft 365 Apps for Enterprise - de-de                    Microsoft.Office                       16.0.15601.20578 16.0.16130.20306 winget<br/>Microsoft 365 Apps for enterprise - en-us                    Microsoft.Office                       16.0.15601.20578 16.0.16130.20306 winget

now I don't need the `de-de` version of Microsoft and want to uninstall it. Is it possible to use winget to purge that software? If I could find a way to get the ID of that package I would use the command

   winget uninstall --id "{<package_ID>}"

to uninstall it. I tried the `winget list Microsoft` and `winget list office` commands, but none shows the specific `Microsoft 365 Apps for Enterprise - de-de` package.

**Comment by mdanish-kh** [comment]
You can uninstall through the full name of the package.

Can you try `winget uninstall 'Microsoft 365 Apps for Enterprise - de-de'`?

**Reply by Foadsf** [reply]
@mdanish-kh I had indeed tried that before, but I get the error message:

> Found a positional argument when none was expected: '365'<br/><br/>Uninstalls the selected package, either found by searching the installed packages list or directly from a manifest. By default, the query must case-insensitively match the id, name, or moniker of the package. Other fields can be used by passing their appropriate option.

**Reply by Trenly** [reply, marked as answer]
> @mdanish-kh I had indeed tried that before, but I get the error message:
> 
> > Found a positional argument when none was expected: '365'Uninstalls the selected package, either found by searching the installed packages list or directly from a manifest. By default, the query must case-insensitively match the id, name, or moniker of the package. Other fields can be used by passing their appropriate option.

You need to quote the name because it has spaces. This error indicates that a space was encountered outside of a quoted string, causing it to believe there is a second argument"

**Reply by Foadsf** [reply]
using double quotation marks as in

    winget uninstall "Microsoft 365 Apps for Enterprise - de-de"

worked. thanks.
